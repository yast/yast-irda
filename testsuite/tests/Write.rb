# encoding: utf-8

#  Write.ycp
#  Test of Irda::Write function
#  Author: Jiri Suchomel <jsuchome@suse.cz>
#  $Id$
module Yast
  class WriteClient < Client
    def main
      # testedfiles: Irda.ycp Service.ycp
      Yast.import "Testsuite"
      Yast.import "Irda"

      @READ = {
        "init"   => {
          "scripts" => {
            "exists"   => true,
            "runlevel" => { "irda" => { "start" => [3] } },
            "comment"  => {}
          }
        },
        "target" => { "stat" => { "isreg" => true } }
      }

      @EX = {
        "target" => {
          "bash_output" => { "exit" => 0 },
          "bash"        => 0,
          "symlink"     => true
        }
      }

      Testsuite.Dump("==== nothing modified: =====================")

      Testsuite.Test(lambda { Irda.Write }, [@READ, {}, @EX], 0)

      Irda.modified = true

      Testsuite.Dump("==== stop irda: ============================")

      Testsuite.Test(lambda { Irda.Write }, [@READ, {}, @EX], 0)

      Testsuite.Dump("==== start irda, create link: ==============")

      Irda.start = true
      Irda.port = "new_port"
      Ops.set(@READ, ["init", "scripts", "runlevel"], {})

      Testsuite.Test(lambda { Irda.Write }, [@READ, {}, @EX], 0)

      nil
    end
  end
end

Yast::WriteClient.new.main
