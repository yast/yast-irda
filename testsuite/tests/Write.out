Dump	==== nothing modified: =====================
Return	true
Dump	==== stop irda: ============================
Write	.sysconfig.irda.IRDA_MAX_BAUD_RATE "0" true
Write	.sysconfig.irda nil true
Execute	.target.bash_output "/bin/systemctl stop irda.service" $["TERM":"raw"] $["exit":0]
Execute	.target.bash "/bin/systemctl is-enabled irda.service" 0
Execute	.target.bash_output "/bin/systemctl  disable irda.service" $["exit":0]
Return	true
Dump	==== start irda, create link: ==============
Write	.sysconfig.irda.IRDA_PORT "new_port" true
Write	.sysconfig.irda.IRDA_MAX_BAUD_RATE "0" true
Write	.sysconfig.irda nil true
Execute	.target.bash_output "/bin/systemctl stop irda.service" $["TERM":"raw"] $["exit":0]
Execute	.target.bash_output "/bin/systemctl start irda.service" $["TERM":"raw"] $["exit":0]
Execute	.target.bash "/bin/systemctl is-enabled irda.service" 0
Return	true
